src/oas.md

```diff
@@ -1612,50 +1647,82 @@ See [Working With Examples](#working-with-examples) for further guidance regardi
 
 This object MAY be extended with [Specification Extensions](#specification-extensions).
 
-##### Media Type Examples
+See also the [Media Type Registry](#media-type-registry).
 
-```json
-{
-  "application/json": {
-    "schema": {
-      "$ref": "#/components/schemas/Pet"
-    },
-    "examples": {
-      "cat": {
-        "summary": "An example of a cat",
-        "value": {
-          "name": "Fluffy",
-          "petType": "Cat",
-          "color": "White",
-          "gender": "male",
-          "breed": "Persian"
-        }
-      },
-      "dog": {
-        "summary": "An example of a dog with a cat's name",
-        "value": {
-          "name": "Puma",
-          "petType": "Dog",
-          "color": "Black",
-          "gender": "Female",
-          "breed": "Mixed"
-        }
-      },
-      "frog": {
-        "$ref": "#/components/examples/frog-example"
-      }
-    }
-  }
-}
+##### Complete vs Streaming Content
+
+The `schema` field MUST be applied to the complete content, as defined by the media type and the context ([Request Body Object](#request-body-object), [Response Object](#response-object), [Parameter Object](#parameter-object), or [Header Object](#header-object).
+Unless some sort of streaming JSON Schema processor is available, this requires loading the entire content into memory.
+This poses a challenge for streamed content, particularly streams where the client is intended to choose when to stop reading as there is no well-defined end to the stream.
+
+###### Binary Streams
+
+The `maxLength` keyword MAY be used to set an expected upper bound on the length of a streaming payload that consists of either string data, including encoded binary data, or to unencoded binary data.
+For unencoded binary, the length is the number of octets.
+For this use case, `maxLength` MAY be implemented outside of regular JSON Schema evaluation as JSON Schema does not directly apply to binary data, and an encoded binary stream may be impractical to store in memory in its entirety.
+
+###### Streaming Sequential Media Types
+
+The `itemSchema` field is provided to support streaming use case for sequential media types.
+Unlike `schema`, which is applied to the complete content (treated as an array as described in the [sequential media types](#sequential-media-types) section), `itemSchema` MUST be applied to each item in the stream independently, which supports processing each item as it is read from the stream.
+
+Both `schema` and `itemSchema` MAY be used in the same Media Type Object, although doing so is unlikely to have significant advantages over using the `items` keyword within the `schema` field.
+OpenAPI Description authors are responsible for avoiding the use of the `schema` in any situation where tooling may not be able to discern when the content is complete.
```

Comment:
I'm not sure I _quite_ follow this. It seems to leap from authors of the description all the way to (client-side?) runtime tooling but... can't the runtime tooling know when it's handling an unterminated stream? Isn't _it_ responsible for knowing that it may not be meaningful to apply `schema` to the content from an unterminated stream?

Reply:
@hudlow it's certainly _possible_ to design tools that do this, but OAS tools aren't monolithic things, and it's not clear to me that that's a reasonable requirement.

@ThomasRooney @disintegrator do either of you have thoughts on this?  @hudlow is correct that it would be nicer if we could make it a requirement.

Reply:
@hudlow after thinking on it more I just removed the caveat.  The requirements as listed (`schema` MUST apply to complete content) already require the error checking you want.  The caveat came from the previous iteration of this PR where the use of `schema` was muddled and ambiguous.

While it's possible that some tool might not handle this correctly, I think we can make that a compliance issue now that I've had a chance to think through it more.  The most likely case where it would go wrong would be some application that, rather than using an actual implementation of OAS just sort of applies bits and pieces of it here and there.  But that's a... creative.. use of the spec that is outside what we need to be concerned about, I think.